pkgs/by-name/cr/creality-print/package.nix

```diff
@@ -0,0 +1,69 @@
+{
+  lib,
+  stdenv,
+  undmg,
+  fetchurl,
+  makeDesktopItem,
+  appimageTools,
+}:
+let
+  pname = "creality-print";
+  version = if stdenv.isLinux then "4.3.7.6627" else "4.3.8.6984";
+  versionChangelog = lib.concatStringsSep "." (lib.take 3 (lib.splitString "." version));
+
+  src = fetchurl {
+    url =
+      let
+        basePart = "https://file2-cdn.creality.com/file";
+        fileHash =
+          if stdenv.isLinux then "05a4538e0c7222ce547eb8d58ef0251e" else "8dcd085c64cc389dacd21cd851593d42";
+        suffix = if stdenv.isLinux then "x86_64-Release.AppImage" else "macx-Release.dmg";
+      in
+      "${basePart}/${fileHash}/Creality_Print-v${version}-${suffix}";
+
+    hash =
+      if stdenv.isLinux then
+        "sha256-WUsL7UbxSY94H4F1Ww8vLsfRyeg2/DZ+V4B6eH3M6+M="
+      else
+        "sha256-0v6/aNzgEf+jF4N+XmPWcpzdJUiUujQSmlcRiGm1z5E=";
+  };
+
+  linuxDerivation = appimageTools.wrapType2 {
+    inherit pname version src;
+    desktopItems = [
+      (makeDesktopItem {
+        name = "creality-print";
+        exec = "creality-print";
+        terminal = false;
+        desktopName = "Creality Print";
+        comment = meta.description;
+        categories = [ "Utility" ];
```

Comment:
any icon?

Reply:
I can't find any examples of other AppImages specifying an icon, how do I do it?

Reply:
https://github.com/NixOS/nixpkgs/pull/326697#discussion_r1676752867

Reply:
It seems there's a `default.png` in the extracted appimage, so we can just use that